src/02_spec/07_modules/cdm/dbgregisters.rst

```diff
@@ -0,0 +1,373 @@
+
+  * - 0x02C1
+    - ``DEBUG_WC_REG0_0``
+    - Bits [15:0] (least significant bits) of Debug Watchpoint Counter 0
+
+  * - 0x02C2
+    - ``DEBUG_WC_REG0_1``
+    - Bits [31:16] of Debug Watchpoint Counter 0
+
+  * - 0x02C3
+    - ``DEBUG_WC_REG1_0``
+    - Bits [15:0] (least significant bits) of Debug Watchpoint Counter 1
+
+  * - 0x02C4
+    - ``DEBUG_WC_REG1_1``
+    - Bits [31:16] of Debug Watchpoint Counter 1
+
+  * - 0x02C5
+    - ``DEBUG_STOP_REG_0``
+    - Bits [15:0] (least significant bits) of Debug Stop Register
+
+  * - 0x02C6
+    - ``DEBUG_STOP_REG_1``
+    - Bits [31:16] of Debug Stop Register
+
+  * - 0x02C7
+    - ``DEBUG_REASON_REG_0``
+    - Bits [15:0] (least significant bits) of Debug Reason Register
+
+  * - 0x02D0
+    - ``DEBUG_REASON_REG_1``
+    - Bits [31:16] of Debug Reason Register
+
+  * - 0x02D1
+    - ``DEBUG_CFG_REG_0``
+    - Bits [15:0] (least significant bits) of Debug Configuration Register
+
+  * - 0x02D2
+    - ``DEBUG_CFG_REG_1``
+    - Bits [31:16] of Debug Configuration Register
+
+  * - 0x02D3
+    - ``DEBUG_WRITE_UPDATE``
+    - Bit 0 of this register will be used to commit write data
+ 
+
+
+Debug Value Registers (``DEBUG_VALUE_REG*_*``)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- Address: *see full register map above*
+- Reset Value: 0
+- Access: read-write
+
+The debug value registers (DVRs) are 32-bit special-purpose supervisor-level registers programmed with the watchpoint/breakpoint addresses or data.
+
+Debug Control Registers (``DEBUG_CTRL_REG*_*``)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- Address: *see full register map above*
+- Reset Value: 0
+- Access: *see below*
+
+The debug control registers (DCRs) are 32-bit special-purpose supervisor-level registers.
+The DCRs are programmed with the watchpoint settings that define how DVRs are compared to the instruction fetch or load/store address or to the load/store data.
+
+.. tabularcolumns:: |p{\dimexpr 0.20\linewidth-2\tabcolsep}|p{\dimexpr 0.40\linewidth-2\tabcolsep}|p{\dimexpr 0.20\linewidth-2\tabcolsep}|p{\dimexpr 0.20\linewidth-2\tabcolsep}|p{\dimexpr 0.20\linewidth-2\tabcolsep}|
+.. flat-table:: DCR Field Description
+  :widths: 2 4 2 2   
+  :header-rows: 1
+
+  * - bits
+    - identifier
+    - reset
+    - Access
+
+  * - 31-8
+    - Reserved
+    - X
+    - R
+
+  * - 7-5
+    - Compare To Condition
+    - 0
+    - R/W
+
+  * - 4
+    - Signed Comparison
+    - 0
+    - R/W
+
+  * - 3-1
+    - Compare Condition
+    - 0
+    - R/W
+
+  * - 0
+    - DVR/DCR Present
+    - 0
+    - R	
+  
+
+Debug Mode Register1 (``DEBUG_MODE_REG1_*``)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- Address: *see full register map above*
+- Reset Value: X (bits: 31-25) | 0 (bits: 23-0)
+- Access: read (bits: 31-25) | read-write (bits: 23-0)
+
+The debug mode register 1 is a 32-bit special-purpose supervisor-level register programmed with the watchpoint/breakpoint settings that define
+how DVR/DCR pairs operate.
```

Comment:
That's a good introductory sentence, but doesn't give necessary information on how that thing actually works. This is a specification -- after reading it, users should know *exactly* what happens when reading or writing this register. You can describe things either by yourself (if it is functionality special to the CDM module), or reference other relevant specifications.

Reply:
These debug registers are specified here: https://openrisc.io/or1k.html#__RefHeading__504809_595890882
I will put this link as the exact reference in the specification. However, I have already mentioned all the relevant details.